components/templates/Books.tsx

```diff
@@ -24,6 +26,18 @@ const useStyles = makeStyles((theme) => ({
   title: {
     marginBottom: theme.spacing(2),
   },
+  linkedBook: {
+    // NOTE: 決め打ち (動画プレイヤーの高さに依存)
+    minHeight: 255,
+  },
+  linkedBookPlaceholder: {
+    // NOTE: 決め打ち (動画プレイヤーの高さに依存)
```

Comment:
> そうですね、あらかじめ決まっているということはコードを読めば明らかなのでなぜなのかという点のみコメントでは表明すれば良いと思います。あるいは「決め打ち」であるということを将来解決したいというニュアンスなのであればTODOとして表明するのが望ましいでしょう。

フィードバックありがとうございます、 @kou029w 

解決したいというモチベーションでいくと、このような内容になるかなと思いました。いかかでしょうか

```suggestion
  linkedBook: {
    // TODO: BookPreview自身によってheightが変動しないことを保証する
    minHeight: 255,
  },
  linkedBookPlaceholder: {
    // TODO: BookPreview自身によってheightが変動しないことを保証する
```

Reply:
「BookPreview自身によってheightが変動しないことを保証する」はBookPreviewの問題のように私は感じたのですがここで解決したい点はそれで正しいですか?
そうではなく将来ここで解決したいこととしては「定数ではなく BookPreview の高さに応じた数値を指定したい」ということではないですか…?
なぜここで高さを指定したかと言うと「提供中のブックの有無に伴う BookPreview の高さ分の上下を抑制するということ」で正しいでしょうか?

Reply:
例えばBookPreviewにheightというpropsが追加され、heightが指定した値で保証されるように作られたら、 `templates/Books` のuseStyleも同じ値でminHeightが制御がすることが可能なのでここにBookPreviewの問題に関心が向いていたのだろうと思います。

言われてみると自分自身論理的飛躍があるようにも思います

Reply:
> そうではなく将来ここで解決したいこととしては「定数ではなく BookPreview の高さに応じた数値を指定したい」ということではないですか…?

暗黙的にBookPreviewの高さが変動する状態は理想的ではないと考えていましたので、自分のイメージにあるものとしては定数による指定は将来的にも残るのであろうと思っていたのではないかなと思います

それがあるべき姿として正しいかというと自信がないところでもあります

> 暗黙的にBookPreviewの高さが変動する状態は理想的ではない

なぜかというと起票できていなかったのですが将来的に[react-window](https://github.com/bvaughn/react-window)や[virtuoso](https://virtuoso.dev/)のようなライブラリの導入の見当はすると考えており、これらの[windowing](https://ja.reactjs.org/docs/optimizing-performance.html#virtualize-long-lists)をおこなうライブラリは一覧するコンポーネントのheightを制御(把握)できているほうが導入しやすい点が念頭にあったためかなとおもいます

Reply:
> なぜここで高さを指定したかと言うと「提供中のブックの有無に伴う BookPreview の高さ分の上下を抑制するということ」で正しいでしょうか?

この点は示していただいたとおりで正しいです

Reply:
少なくともBooksにおけるレイアウトに関するコメントでBookPreviewに関するTODOを残すのは、単純にコメントを残す場所としては適切ではないことは理解しました。

(私の視点において)この場所に適切なコメントを残すのはNOTE(なぜそうしたのか)で残す方が議論の余地が生まれににくいかもしれません。

```suggestion
  linkedBook: {
    // NOTE: BookPreviewに依存する&事前に高さを確定できないので決め打ち
    minHeight: 255,
  },
  linkedBookPlaceholder: {
    // NOTE: BookPreviewに依存する&事前に高さを確定できないので決め打ち
```

Reply:
散漫な思考の垂れ流しになりお手数かけました :sweat_drops: & お付き合いくださりありがとうございます :pray: